src/epr/models.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+# SPDX-FileCopyrightText: © 2024 Brett Smith <xbcsmith@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+from dataclasses import asdict, dataclass, field
+from enum import Enum
+from typing import Any, Dict, List
+
+from .fingerprint import GroupFingerprint, ReceiverFingerprint
+
+
+class ModelType(Enum):
+    EVENT = "Event"
+    RECEIVER = "Event_Receiver"
+    GROUP = "Event_Receiver_Group"
+
+    def lower(self):
+        return self.value.lower()
+
+    def lower_plural(self):
+        return self.lower() + "s"
+
+
+@dataclass
+class Model:
+    """Base class for data objects. Provides as_dict"""
+
+    def as_dict(self):
+        """Get a dictionary contain object properties"""
+        return asdict(self)
+
+    def as_dict_query(self):
+        """Get a dictionary contain object properties"""
+        return {k: v for k, v in self.as_dict().items() if v}
+
+
+@dataclass
+class Event(Model):
+    """Data class for Events"""
+
+    id: str = field(default="", compare=False)
+    name: str = ""
+    version: str = ""
+    release: str = ""
+    platform_id: str = ""
+    package: str = ""
+    description: str = ""
+    payload: Dict[str, Any] = field(default_factory=dict)
+    success: bool = False
+    created_at: str = field(default="", compare=False)
+    event_receiver_id: str = ""
+    event_receiver: Dict[str, Any] = field(default_factory=dict)
+
+
+@dataclass
+class Event_Receiver(Model):
```

Comment:
By convention should be PascalCase. Also applies to `Event_Receiver_Group` model

Reply:
Fixed